### PR TITLE
add verify-conversions target

### PR DESCRIPTION
**Pull request by sonasingh46**
/kind cleanup

**What this PR does / why we need it**:
This PR adds the following make targets: 
1. `make verify-conversions`: Verifies that the expected conversions are in place.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Reference issue #2026 

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
```release-note
None
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
@sonasingh46 looks like this might overlap with #2040 (there was another issue opened a while ago about missing conversion func and I guess you and @mboersma both started working on it at the same time through different issues)

**Comment by sonasingh46** [comment]
@CecileRobertMichon -- Sorry, I was not aware. This PR has some Makefile changes too. Either @mboersma can pick those from here in his same PR and then I will close this PR. Or once his PR gets merged, I can rebase and can get the Makefile changes in.

**Comment by mboersma** [comment]
Ha! No, this is entirely similar to my PR, except that I added in a bunch of unrelated code cleanup. I'll close my PR and review this one instead, thanks @sonasingh46!

**Comment by mboersma** [comment]
This PR fixes #1989.

**Review by mboersma — APPROVED** [review]
/lgtm

I tested it locally as well and everything looks good.

File: hack/tools/tools.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This package imports things required by build scripts, to force `go mod` to see them as dependencies
+package tools
+
+import (
+	_ "sigs.k8s.io/cluster-api/hack/tools"
```
Comment (CecileRobertMichon) [inline hack/tools/tools.go:24]: do we need this? why not install it with go install how we install others like envsubst?

Comment (mboersma) [reply hack/tools/tools.go:24]: The CAPI hack/tools package has a `replace` statement in it with a fake SHA that prevents `go install` (or `go get`) from working. Adding a local package in CAPZ and a local go.mod seems like the only solution, at least that's where I got to as well when looking at this.

**Pull request merged**